core/src/main/java/hudson/FilePath.java

```diff
@@ -2995,8 +2996,17 @@ private void writeObject(ObjectOutputStream oos) throws IOException {
         oos.writeBoolean(channel==null);
     }
 
+    private Channel _getChannelForSerialization() {
+        try {
+            return getChannelForSerialization();
+        } catch (NotSerializableException x) {
+            LOGGER.log(Level.WARNING, "A FilePath object is being serialized when it should not be, indicating a bug in a plugin. See https://jenkins.io/redirect/filepath-serialization for details.", x);
```

Comment:
https://github.com/jenkins-infra/jenkins.io/pull/2233, to begin with just pointing to the existing JIRA issue.